docs/using-hive.md

```diff
@@ -295,6 +296,110 @@ metadata:
+
+#### Creating a Secret for Credentials
+The Nutanix credentials must be stored as a Kubernetes secret in the namespace where Hive operates. Create a secret with the following format:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: nutanix-cloud-credentials
+  namespace: hive
+type: Opaque
+data:
+  username: <base64-encoded-username>
+  password: <base64-encoded-password>
+```
+
+To create the secret using `oc`, first encode the values in Base64:
+
+```sh
+echo -n "<value>" | base64
+```
+
+Then, apply the secret using:
+
+```sh
+oc apply -f nutanix-cloud-credentials.yaml
+```
+
+#### Additional Required Secrets
+In addition to the Hive credentials, OpenShift requires additional secrets in specific namespaces for authentication with Nutanix.
+
+##### Secret for OpenShift Machine API
+This secret is required by the OpenShift Machine API to manage machines on Nutanix:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+   name: nutanix-credentials
+   namespace: openshift-machine-api 
+type: Opaque
+stringData:
+  credentials: |
+    [{"type":"basic_auth","data":{"prismCentral":{"username":"${NUTANIX_USERNAME}","password":"${NUTANIX_PASSWORD}"}}}]
+```
+
+##### Secret for OpenShift Cloud Controller Manager
+This secret is required by the OpenShift Cloud Controller Manager to integrate OpenShift with Nutanix infrastructure:
+
+```yamlz
+apiVersion: v1
+kind: Secret
+metadata:
+   name: nutanix-credentials
+   namespace: openshift-cloud-controller-manager
+type: Opaque
+stringData:
+  credentials: |
+    [{"type":"basic_auth","data":{"prismCentral":{"username":"${NUTANIX_USERNAME}","password":"${NUTANIX_PASSWORD}"}}}]
+```
+
+#### Why These Secrets Are Required
+- **Hive Secret**: Used by Hive for provisioning clusters and managing resources.
+- **Machine API Secret**: Required for the OpenShift Machine API to manage and create worker nodes on Nutanix.
+- **Cloud Controller Manager Secret**: Enables OpenShift to interact with Nutanix for networking, load balancing, and other infrastructure-related tasks.
+
+Each of these secrets plays a critical role in ensuring a seamless integration between OpenShift and Nutanix, allowing for automated cluster deployment and lifecycle management.
+
+#### Using the Secret in Hive
+Once the secret is created, reference it in the `ClusterDeployment` or `ClusterPool` configuration:
+
+```yaml
+spec:
+  platform:
+    nutanix:
+      credentialsSecretRef:
+        name: nutanix-cloud-credentials
+```
+
+This ensures that Hive can retrieve the necessary credentials to interact with Nutanix for cluster provisioning.
+
+#### Additional Considerations
+- Ensure the Nutanix user has appropriate permissions to create virtual machines and network resources.
+- Verify network connectivity between the OpenShift cluster nodes and Nutanix infrastructure.
+- If using self-signed certificates in Nutanix, additional TLS configurations may be required.
```

Comment:
This can be beefed up now I think. In particular, it would be good to describe the interaction between install-config's additionalTrustBundle[Policy] and cd.spec.platform.nutanix.certificatesSecretRef. Namely:
- The former is for day 0 and the latter is (or will be) for day 2. (Uh, I forget if that's true -- does provisioning still work if you *only* specify the former?)
- As currently written, you can specify both if you want/need them to be different; otherwise the latter will be used for both day 0 and day 2 stuff.

Also I think it doesn't matter if the certs are self-signed. I would probably change the first sentence to
"If the Nutanix Prism Central uses certificates..."